### DISCUSSION TITLE
How can I avoid MDX converting any symbols in math blocks ($$...$$) while not using remark-math nor rehype-katex?

### DISCUSSION THREAD

**Discussion by rinarakaki** [mdx-js/mdx#1386] [Q&A] [answered]
Hello team,

I'm struggling to render a valid sequence to be read by MathJax with MDX.
I first thought symbols between  `$$` wouldn't be touched by MDX at all but I was wrong.
So how can I configure MDX not to change anything inside those symbols?

*I have no emark-math nor rehype-katex installed because I'm not intending to use them.

**Comment by ChristianMurphy** [comment, marked as answer]
@rnarkk could you describe your goal?
Do you want to show the raw LaTeX formula? Or do you want to typeset the math in the document later using a different library than `rehype-katex`? Something else?

**Reply by rinarakaki** [reply]
My goal is to show a LaTeX formula on HTML so that MathJax can read it after the whole process (MDX > JSX > HTML).

For example, when I write this in a MDX file, those `_`s will be interpreted as `<em>`s.
`$$e_{i_1, i_2} e_{i_2, i_1} = e_{i_1, i_1}$$`

I cannot escape every single `_` or any other symbols that have meanings in Markdown because I will have to write countless formulas.

**Reply by ChristianMurphy** [reply]
No matter what you use to process the math for typesetting, you're going to want `remark-math` to extract the `$$` blocks and `$` inline equations to prevent them from being interpreted as plain markdown.
For the specific way you want to add the content to the page, that is possible through plugins, there isn't an existing one (that I'm aware of) that processes in the exact way you want.
You can create your own plugin (https://unifiedjs.com/learn) that takes the `math` and `inlineMath` nodes and creates the HTML (hast) nodes that you want. https://mdxjs.com/advanced/transform-content.

Or you could use `rehype-mathjax` :sweat_smile:

**Reply by rinarakaki** [reply]
Thank you very much for providing an informative direction.
